### ISSUE TITLE
PromptProductPurchaseFinished is depreciated, but PromptPurchaseFinished does not fire for Developer products

### ISSUE THREAD

**Issue by Quenty** [Roblox/Core-Scripts#878]
The event `PromptProductPurchaseFinished` is depreciated, but `PromptPurchaseFinished` does not fire for Developer products. This means that `PromptProductPurchaseFinished` must be used to detect products finishing purchase, forcing depreciated code use. 

The code causing it:
https://github.com/Roblox/Core-Scripts/blob/0376349c8547c027de58a4e1dd1ffd12051292cf/CoreScriptsRoot/CoreScripts/PurchasePromptScript3.lua#L945

* http://wiki.roblox.com/index.php?title=API:Class/MarketplaceService/PromptProductPurchaseFinished
* http://wiki.roblox.com/index.php?title=API:Class/MarketplaceService/PromptPurchaseFinished

**Comment by blujagu** [comment]
MarketplaceService.ProcessReceipt is the replacement for detecting purchase completion, so your issue should be covered. However, this is a major issue for a properly constructed UI that wants to disable it's "Buy" buttons until the server returns something. **If a player presses "Cancel", the ONLY event to listen for is PromptProductPurchaseFinished, so it being deprecated is not a good thing until there is a fitting replacement for capturing the cancel button event.**

**Comment by Quenty** [comment]
Yes. To clarify:`MarketplaceService.ProcessReceipt` only is called when a purchase is _made_ (i.e. the user has bought the product). I need to know when a purchase is not made too.

The solution is to make PromptPurchaseFinished fire on DeveloperProducts when cancelled.

**Comment by ArceusInator** [comment]
![dlzapoo](https://user-images.githubusercontent.com/2127060/28246469-7c04d730-69cf-11e7-93e9-c59856529191.png)
![dlzapoo](https://user-images.githubusercontent.com/2127060/28246469-7c04d730-69cf-11e7-93e9-c59856529191.png)
![dlzapoo](https://user-images.githubusercontent.com/2127060/28246469-7c04d730-69cf-11e7-93e9-c59856529191.png)
![dlzapoo](https://user-images.githubusercontent.com/2127060/28246469-7c04d730-69cf-11e7-93e9-c59856529191.png)
![dlzapoo](https://user-images.githubusercontent.com/2127060/28246469-7c04d730-69cf-11e7-93e9-c59856529191.png)
![dlzapoo](https://user-images.githubusercontent.com/2127060/28246469-7c04d730-69cf-11e7-93e9-c59856529191.png)
![dlzapoo](https://user-images.githubusercontent.com/2127060/28246469-7c04d730-69cf-11e7-93e9-c59856529191.png)
![dlzapoo](https://user-images.githubusercontent.com/2127060/28246469-7c04d730-69cf-11e7-93e9-c59856529191.png)
![dlzapoo](https://user-images.githubusercontent.com/2127060/28246469-7c04d730-69cf-11e7-93e9-c59856529191.png)
![dlzapoo](https://user-images.githubusercontent.com/2127060/28246469-7c04d730-69cf-11e7-93e9-c59856529191.png)
![dlzapoo](https://user-images.githubusercontent.com/2127060/28246469-7c04d730-69cf-11e7-93e9-c59856529191.png)
![dlzapoo](https://user-images.githubusercontent.com/2127060/28246469-7c04d730-69cf-11e7-93e9-c59856529191.png)
![dlzapoo](https://user-images.githubusercontent.com/2127060/28246469-7c04d730-69cf-11e7-93e9-c59856529191.png)
![dlzapoo](https://user-images.githubusercontent.com/2127060/28246469-7c04d730-69cf-11e7-93e9-c59856529191.png)
![dlzapoo](https://user-images.githubusercontent.com/2127060/28246469-7c04d730-69cf-11e7-93e9-c59856529191.png)
![dlzapoo](https://user-images.githubusercontent.com/2127060/28246469-7c04d730-69cf-11e7-93e9-c59856529191.png)
![dlzapoo](https://user-images.githubusercontent.com/2127060/28246469-7c04d730-69cf-11e7-93e9-c59856529191.png)
![dlzapoo](https://user-images.githubusercontent.com/2127060/28246469-7c04d730-69cf-11e7-93e9-c59856529191.png)
![dlzapoo](https://user-images.githubusercontent.com/2127060/28246469-7c04d730-69cf-11e7-93e9-c59856529191.png)
![dlzapoo](https://user-images.githubusercontent.com/2127060/28246469-7c04d730-69cf-11e7-93e9-c59856529191.png)
![dlzapoo](https://user-images.githubusercontent.com/2127060/28246469-7c04d730-69cf-11e7-93e9-c59856529191.png)
![dlzapoo](https://user-images.githubusercontent.com/2127060/28246469-7c04d730-69cf-11e7-93e9-c59856529191.png)
![dlzapoo](https://user-images.githubusercontent.com/2127060/28246469-7c04d730-69cf-11e7-93e9-c59856529191.png)
![dlzapoo](https://user-images.githubusercontent.com/2127060/28246469-7c04d730-69cf-11e7-93e9-c59856529191.png)
![dlzapoo](https://user-images.githubusercontent.com/2127060/28246469-7c04d730-69cf-11e7-93e9-c59856529191.png)
![dlzapoo](https://user-images.githubusercontent.com/2127060/28246469-7c04d730-69cf-11e7-93e9-c59856529191.png)
![dlzapoo](https://user-images.githubusercontent.com/2127060/28246469-7c04d730-69cf-11e7-93e9-c59856529191.png)
![dlzapoo](https://user-images.githubusercontent.com/2127060/28246469-7c04d730-69cf-11e7-93e9-c59856529191.png)
![dlzapoo](https://user-images.githubusercontent.com/2127060/28246469-7c04d730-69cf-11e7-93e9-c59856529191.png)
![dlzapoo](https://user-images.githubusercontent.com/2127060/28246469-7c04d730-69cf-11e7-93e9-c59856529191.png)
![dlzapoo](https://user-images.githubusercontent.com/2127060/28246469-7c04d730-69cf-11e7-93e9-c59856529191.png)
 Hey ME Cum IN UR Aasrse.

**Comment by boynedmaster** [comment]
Feel the same way as @blujagu. I have a whole comment somewhere in my code cursing Roblox for deprecating something it when there's no other function to detect if a user cancels.

**Comment by cliffchapmanrbx** [comment]
Hi there, thank you for your feedback. We're going to be closing this repository as we've chosed [the Roblox Dev Forum](https://devforum.roblox.com/) as our venue for developer feedback. Please feel free to migrate your feedback to a DevForum post if you feel it is still relevant.
